fix: LEAP-947: Check task when we get a response from ML

hlomzik opened the pull request:
Add a wrapper to check that we are going to apply response from ML-backend to the same task we sent a request from in a first place.

### PR fulfills these requirements
- [ ] Tests for the changes have been added/updated
- [ ] Docs have been added/updated
- [x] Best efforts were made to ensure docs/code are concise and coherent (checked for spelling/grammatical errors, commented out code, debug logs etc.)
- [x] Self-reviewed and ran all changes on a local instance

### Describe the reason for change
If ML-backend responds long enough and we switched to another task (by submit, prev, postpone, etc) then its response will be applied to a wrong annotation, not only making a confusion but potentially distorting the image because of image sizes embedded into regions.

### What alternative approaches were there?
We should also stop requests when we leave a task, this will be done in a future.

### This change affects (describe how if yes)
- [ ] Performance
- [ ] Security
- [ ] UX

### Does this PR introduce a breaking change?
- [ ] Yes, and covered entirely by feature flag(s)
- [ ] Yes, and covered partially by feature flag(s)
- [ ] No
- [x] Fixes breaking change (data corruption)

### What level of testing was included in the change?
- [ ] e2e (codecept)
- [ ] integration (cypress)
- [ ] unit (jest)

hlomzik committed:
fix: LEAP-947: Check task when we get a response from ML

Add a wrapper to check that we are going to apply response from ML-backend
to the same task we sent a request from in a first place.

robot-ci-heartex committed:
ci: Build frontend

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/8723060864

yyassi-heartex commented on web/libs/editor/src/stores/AppStore.js:
```diff
@@ -798,11 +798,18 @@ export default types
+        }
+      } catch(e) {
         self.setFlags({ awaitingSuggestions: false });
+        // @todo handle errors + situation when task is changed
```
since there is a todo here it might be prudent to add a `console.error` temporarily as a placeholder while a more elegant solution is put in place
```suggestion
        // @todo handle errors + situation when task is changed
        console.error('Error:', e);
```

hlomzik replied:
state here is not always an error like in this specific case we are fixing here — it's just a way to ignore the wrong request. later we'll come up with two different things:
- proper handling for errors from ML backend
- proper request management and cancellation

yyassi-heartex replied:
ok sounds good

yyassi-heartex commented on web/libs/editor/src/stores/AppStore.js:
```diff
@@ -798,11 +798,18 @@ export default types
+      } catch(e) {
         self.setFlags({ awaitingSuggestions: false });
+        // @todo handle errors + situation when task is changed
+        return;
```
maybe im misreading/misunderstanding the code but is this return needed? doesn't really appear to be giving us an early return. Am i missing something?
```suggestion
```

hlomzik replied:
ah, yeah, forgot to remove it after rearranging the code, thank you!

yyassi-heartex commented on web/apps/labelstudio/src/pages/DataManager/DataManager.jsx:
```diff
@@ -140,7 +140,20 @@ export const DataManagerPage = ({ ...props }) => {
+          if (task.id === dataManager.lsf.task.id) {
+            resolve(response);
+          } else {
+            reject();
```
if we `reject()` then we move on to `lsf.loadSuggestions(wrappedRequest, (response) => {` wouldn't response be `undefined`? if thats the case line 157 `if (response.data) {` will throw an error most likely

yyassi-heartex replied:
ahh, disregard - we catch in loadSuggestions so we dont fire `dataParser`. Resolving convo

hlomzik committed:
Remove excess return

Co-authored-by: yyassi-heartex <104568407+yyassi-heartex@users.noreply.github.com>

robot-ci-heartex committed:
ci: Build frontend

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/8723717356

hlomzik commented:
/git merge
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/8729250192)
> Successfully merged:  5 files changed, 41 insertions(+), 1 deletion(-)

hlomzik committed:
Merge branch 'develop' into 'fb-leap-947/wrong-task-suggestions'

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/8729250192

robot-ci-heartex committed:
ci: Build frontend

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/8729274633

hlomzik commented:
/git merge
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/8792350026)
> Successfully merged:  53 files changed, 1618 insertions(+), 898 deletions(-)

hlomzik committed:
Merge branch 'develop' into 'fb-leap-947/wrong-task-suggestions'

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/8792350026

robot-ci-heartex committed:
ci: Build frontend

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/8792369736
